lib/exports.js

```diff
@@ -4,9 +4,45 @@ var util = require('util');
 var swaggerRouter = require('swagger-router');
 var yaml = require('js-yaml');
 var fs = require('fs');
+var EventEmitter = require('events').EventEmitter;
 
 var exports = {};
 
+function Lifecycle() {
+    EventEmitter.call(this);
+    this._initialized = false;
+}
+util.inherits(Lifecycle, EventEmitter);
+
+/**
+ * Lazily sets up forwarding of the application-lifecycle events
+ * to the exported object. Only called if the client subscribed to
+ * some events.
+ *
+ * @private
+ */
+Lifecycle.prototype._setup = function() {
+    var self = this;
+    // The server will certainly be initialized by now
+    var server = require('./server').server;
+    server.on('close', function() { self.emit('close'); });
+};
+[
+    'addListener',
+    'on',
+    'once',
+    'prependListener',
+    'prependOnceListener'
```

Comment:
`removeListener`should be in the list too. It's not relevant right now as there is no sense to remove a listener at the end of execution, but let's have it there for future uses.